packages/windows_application_windows.go

```diff
@@ -0,0 +1,122 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package packages
+
+import (
+	//"golang.org/x/sys/windows"
+	//"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+	//"time"
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/GoogleCloudPlatform/osconfig/clog"
+	"golang.org/x/sys/windows/registry"
+)
+
+func getStringValueReturnEmptyIfError(k *registry.Key, value string) string {
```

Comment:
GetStringValue already returns "" on any error, this is actually best practice with Go functions. You should be able to remove the need for this function and just ignore error (make sure to add a comment as to why)

Reply:
This function exists because it returns one value so I can use it directly when creating WindowsApplication structure like that:

```
    return &WindowsApplication{
      DisplayName:    displayName,
      DisplayVersion: getStringValueReturnEmptyIfError(k, "DisplayVersion"),
      Publisher:      getStringValueReturnEmptyIfError(k, "Publisher"),
      InstallDate:    parseDate(getStringValueReturnEmptyIfError(k, "InstallDate")),
      HelpLink:       getStringValueReturnEmptyIfError(k, "HelpLink"),
    }  
```

Reply:
That's overly complicated and introduces an addition nil check for no good reason, just do it this way:

```
displayVersion, _, _ := k.GetStringValue("DisplayVersion"))
publisher, _, _ := k.GetStringValue("Publisher"))
return &WindowsApplication{ 
  DisplayName: displayName, 
  DisplayVersion: displayVersion,
  Publisher: publisher,
  ...
 }
```